### DISCUSSION TITLE
F401 - First unused import not caught

### DISCUSSION THREAD

**Discussion by AbdealiLoKo** [astral-sh/ruff#10133] [Q&A] [answered]
Opening this as a discussion to understand if this is a bug ?

I had a case where I had a file with something like this:
```py
$ cat run.py
from __future__ import annotations

import myapp.abc.qwe
import myapp.abc.xyz
```

And when I ran ruff it gave:
```
$ venv/bin/ruff run.py
run.py:
  4:8 F401 [*] `myapp.abc.xyz` imported but unused

Found 1 error.
[*] 1 fixable with the `--fix` option.
```

In my opinion, this should have given me 2 lint errors: 1 for the `qwe` import which is unused. And second for the `xyz` import that is unused

But it looks like ruff is assuming that the first line is needed and being used ? Which seemed odd

- Python version: Python 3.7.16
- Ruff version: ruff 0.2.2

**Comment by JonathanPlasse** [comment, marked as answer]
- There is an issue tracking this problem #60
